### DISCUSSION TITLE
clang-11: error

### DISCUSSION THREAD

**Discussion by xchengood** [idaholab/moose#20704] [Q&A General] [answered]
Hi Moose experts or users,

I don't know what reason triggers the following errors. Before I update Xcode to the newest version. But that error occurs. So I just downgrade Xcode to 13.1. But errors are still there. Any thoughts to solve it. Thank you.

> (moose) xinchen@x86_64-apple-darwin13 tensor_mechanics % cd ~/projects/moose/modules/tensor_mechanics
make -j4
Using HIT from /Users/xinchen/projects/moose/framework/contrib/hit
Compiling C++ (in opt mode) /Users/xinchen/projects/moose/modules/tensor_mechanics/build/unity_src/materials_crystal_plasticity_Unity.C...
In file included from /Users/xinchen/projects/moose/modules/tensor_mechanics/build/unity_src/materials_crystal_plasticity_Unity.C:2:
In file included from /Users/xinchen/projects/moose/modules/tensor_mechanics/src/materials/crystal_plasticity/ComputeCrystalPlasticityEigenstrainBase.C:10:
In file included from /Users/xinchen/projects/moose/modules/tensor_mechanics/build/header_symlinks/ComputeCrystalPlasticityEigenstrainBase.h:12:
In file included from /Users/xinchen/projects/moose/modules/tensor_mechanics/build/header_symlinks/ComputeEigenstrainBase.h:12:
In file included from /Users/xinchen/projects/moose/framework/build/header_symlinks/Material.h:13:
In file included from /Users/xinchen/projects/moose/framework/build/header_symlinks/MaterialBase.h:13:
In file included from /Users/xinchen/projects/moose/framework/build/header_symlinks/MooseObject.h:13:
In file included from /Users/xinchen/projects/moose/framework/build/header_symlinks/InputParameters.h:13:
In file included from /Users/xinchen/projects/moose/framework/build/header_symlinks/MooseUtils.h:13:
In file included from /Users/xinchen/projects/moose/framework/build/header_symlinks/HashMap.h:13:
In file included from /Users/xinchen/miniconda3/envs/moose/libmesh/include/libmesh/threads.h:24:
In file included from /Users/xinchen/miniconda3/envs/moose/libmesh/include/libmesh/libmesh_common.h:36:
In file included from /Users/xinchen/miniconda3/envs/moose/bin/../include/c++/v1/cstdlib:85:
/Users/xinchen/miniconda3/envs/moose/bin/../include/c++/v1/stdlib.h:142:34: error: unknown type name 'ldiv_t'
inline _LIBCPP_INLINE_VISIBILITY ldiv_t div(long __x, long __y) _NOEXCEPT {
                                 ^
/Users/xinchen/miniconda3/envs/moose/bin/../include/c++/v1/stdlib.h:143:12: error: no member named 'ldiv' in the global namespace
  return ::ldiv(__x, __y);
         ~~^
/Users/xinchen/miniconda3/envs/moose/bin/../include/c++/v1/stdlib.h:146:34: error: unknown type name 'lldiv_t'
inline _LIBCPP_INLINE_VISIBILITY lldiv_t div(long long __x,
                                 ^
/Users/xinchen/miniconda3/envs/moose/bin/../include/c++/v1/stdlib.h:148:12: error: no member named 'lldiv' in the global namespace
  return ::lldiv(__x, __y);
         ~~^
In file included from /Users/xinchen/projects/moose/modules/tensor_mechanics/build/unity_src/materials_crystal_plasticity_Unity.C:2:
In file included from /Users/xinchen/projects/moose/modules/tensor_mechanics/src/materials/crystal_plasticity/ComputeCrystalPlasticityEigenstrainBase.C:10:
In file included from /Users/xinchen/projects/moose/modules/tensor_mechanics/build/header_symlinks/ComputeCrystalPlasticityEigenstrainBase.h:12:
In file included from /Users/xinchen/projects/moose/modules/tensor_mechanics/build/header_symlinks/ComputeEigenstrainBase.h:12:
In file included from /Users/xinchen/projects/moose/framework/build/header_symlinks/Material.h:13:
In file included from /Users/xinchen/projects/moose/framework/build/header_symlinks/MaterialBase.h:13:
In file included from /Users/xinchen/projects/moose/framework/build/header_symlinks/MooseObject.h:13:
In file included from /Users/xinchen/projects/moose/framework/build/header_symlinks/InputParameters.h:13:
In file included from /Users/xinchen/projects/moose/framework/build/header_symlinks/MooseUtils.h:13:
In file included from /Users/xinchen/projects/moose/framework/build/header_symlinks/HashMap.h:13:
In file included from /Users/xinchen/miniconda3/envs/moose/libmesh/include/libmesh/threads.h:24:
In file included from /Users/xinchen/miniconda3/envs/moose/libmesh/include/libmesh/libmesh_common.h:36:
/Users/xinchen/miniconda3/envs/moose/bin/../include/c++/v1/cstdlib:99:9: error: no member named 'size_t' in the global namespace
using ::size_t;
      ~~^
/Users/xinchen/miniconda3/envs/moose/bin/../include/c++/v1/cstdlib:100:9: error: no member named 'div_t' in the global namespace
using ::div_t;
      ~~^
/Users/xinchen/miniconda3/envs/moose/bin/../include/c++/v1/cstdlib:101:9: error: no member named 'ldiv_t' in the global namespace
using ::ldiv_t;
      ~~^
/Users/xinchen/miniconda3/envs/moose/bin/../include/c++/v1/cstdlib:103:9: error: no member named 'lldiv_t' in the global namespace
using ::lldiv_t;
      ~~^
/Users/xinchen/miniconda3/envs/moose/bin/../include/c++/v1/cstdlib:105:9: error: no member named 'atof' in the global namespace
using ::atof;
      ~~^
/Users/xinchen/miniconda3/envs/moose/bin/../include/c++/v1/cstdlib:106:9: error: no member named 'atoi' in the global namespace
using ::atoi;
      ~~^
/Users/xinchen/miniconda3/envs/moose/bin/../include/c++/v1/cstdlib:107:9: error: no member named 'atol' in the global namespace
using ::atol;
      ~~^
/Users/xinchen/miniconda3/envs/moose/bin/../include/c++/v1/cstdlib:109:9: error: no member named 'atoll' in the global namespace
using ::atoll;
      ~~^
/Users/xinchen/miniconda3/envs/moose/bin/../include/c++/v1/cstdlib:111:9: error: no member named 'strtod' in the global namespace
using ::strtod;
      ~~^
/Users/xinchen/miniconda3/envs/moose/bin/../include/c++/v1/cstdlib:112:9: error: no member named 'strtof' in the global namespace
using ::strtof;
      ~~^
/Users/xinchen/miniconda3/envs/moose/bin/../include/c++/v1/cstdlib:113:9: error: no member named 'strtold' in the global namespace
using ::strtold;
      ~~^
/Users/xinchen/miniconda3/envs/moose/bin/../include/c++/v1/cstdlib:114:9: error: no member named 'strtol' in the global namespace
using ::strtol;
      ~~^
/Users/xinchen/miniconda3/envs/moose/bin/../include/c++/v1/cstdlib:116:9: error: no member named 'strtoll' in the global namespace
using ::strtoll;
      ~~^
/Users/xinchen/miniconda3/envs/moose/bin/../include/c++/v1/cstdlib:118:9: error: no member named 'strtoul' in the global namespace
using ::strtoul;
      ~~^
/Users/xinchen/miniconda3/envs/moose/bin/../include/c++/v1/cstdlib:120:9: error: no member named 'strtoull' in the global namespace
using ::strtoull;
      ~~^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
PLEASE submit a bug report to https://bugs.llvm.org/ and include the crash backtrace, preprocessed source, and associated run script.
Stack dump:
0.	Program arguments: x86_64-apple-darwin13.4.0-clang++ -march=core2 -mtune=haswell -std=gnu++17 -DNDEBUG -march=core2 -mtune=haswell -mssse3 -ftree-vectorize -fPIC -fstack-protector-strong -O2 -pipe -stdlib=libc++ -fvisibility-inlines-hidden -std=c++14 -fmessage-length=0 -isystem /Users/xinchen/miniconda3/envs/moose/include -O2 -felide-constructors -Qunused-arguments -Wunused-parameter -Wunused -fopenmp -DMETHOD=opt -Werror=return-type -Werror=reorder -Woverlength-strings -Wall -Wextra -DADFPARSER_INCLUDES="-I/Users/xinchen/projects/moose/framework/include/utils -I/Users/xinchen/projects/moose/framework/include/base -I/Users/xinchen/miniconda3/envs/moose/libmesh/include" -DTENSOR_MECHANICS_ENABLED -I/Users/xinchen/projects/moose/framework/build/header_symlinks -I/Users/xinchen/projects/moose/framework/contrib/boost/include -I/Users/xinchen/projects/moose/framework/contrib/json/include -I/Users/xinchen/projects/moose/framework/contrib/json/include/nlohmann -I/Users/xinchen/projects/moose/framework/contrib/mtwist/include -I/Users/xinchen/projects/moose/framework/contrib/pcre/include -I/Users/xinchen/projects/moose/framework/contrib/pugixml/include -I/Users/xinchen/projects/moose/framework/contrib/tinydir/include -I/Users/xinchen/projects/moose/framework/contrib/gtest -I/Users/xinchen/projects/moose/framework/contrib/hit -I/Users/xinchen/miniconda3/envs/moose/libmesh/include -I/Users/xinchen/miniconda3/envs/moose/libmesh-vtk/include/vtk-6.3 -D_THREAD_SAFE -pthread -I/Users/xinchen/miniconda3/envs/moose/include -I/Users/xinchen/miniconda3/envs/moose//include -I/Users/xinchen/projects/moose/modules/tensor_mechanics/build/header_symlinks -I/Users/xinchen/miniconda3/envs/moose/libmesh/include -I/Users/xinchen/miniconda3/envs/moose/libmesh-vtk/include/vtk-6.3 -D_THREAD_SAFE -pthread -I/Users/xinchen/miniconda3/envs/moose/include -I/Users/xinchen/miniconda3/envs/moose//include -MMD -MP -MF /Users/xinchen/projects/moose/modules/tensor_mechanics/build/unity_src/materials_crystal_plasticity_Unity.x86_64-apple-darwin13.4.0.opt.lo.d -MT /Users/xinchen/projects/moose/modules/tensor_mechanics/build/unity_src/materials_crystal_plasticity_Unity.x86_64-apple-darwin13.4.0.opt.lo -c /Users/xinchen/projects/moose/modules/tensor_mechanics/build/unity_src/materials_crystal_plasticity_Unity.C -fno-common -DPIC -o /Users/xinchen/projects/moose/modules/tensor_mechanics/build/unity_src/.libs/materials_crystal_plasticity_Unity.x86_64-apple-darwin13.4.0.opt.o -I/Users/xinchen/miniconda3/envs/moose/include 
1.	<eof> parser at end of file
0  libLLVM-11.dylib         0x0000000112cfac16 llvm::sys::PrintStackTrace(llvm::raw_ostream&) + 54
1  libLLVM-11.dylib         0x0000000112cf9a08 llvm::sys::RunSignalHandlers() + 248
2  libLLVM-11.dylib         0x0000000112cfa142 llvm::sys::CleanupOnSignal(unsigned long) + 258
3  libLLVM-11.dylib         0x0000000112c2224e CrashRecoverySignalHandler(int) + 206
4  libsystem_platform.dylib 0x00007ff812f04e2d _sigtramp + 29
5  libclang-cpp.11.dylib    0x000000010b559075 clang::EvaluatedExprVisitorBase<std::__1::add_pointer, (anonymous namespace)::DeferredDiagnosticsEmitter>::VisitStmt(clang::Stmt*) + 133
6  libclang-cpp.11.dylib    0x000000010af7ec5f clang::ASTContext::getBaseElementType(clang::QualType) const + 47
7  libclang-cpp.11.dylib    0x000000010b55856a clang::StmtVisitorBase<std::__1::add_pointer, (anonymous namespace)::DeferredDiagnosticsEmitter, void>::Visit(clang::Stmt*) + 1418
8  libclang-cpp.11.dylib    0x000000010b559075 clang::EvaluatedExprVisitorBase<std::__1::add_pointer, (anonymous namespace)::DeferredDiagnosticsEmitter>::VisitStmt(clang::Stmt*) + 133
9  libclang-cpp.11.dylib    0x000000010b559075 clang::EvaluatedExprVisitorBase<std::__1::add_pointer, (anonymous namespace)::DeferredDiagnosticsEmitter>::VisitStmt(clang::Stmt*) + 133
10 libclang-cpp.11.dylib    0x000000010b559075 clang::EvaluatedExprVisitorBase<std::__1::add_pointer, (anonymous namespace)::DeferredDiagnosticsEmitter>::VisitStmt(clang::Stmt*) + 133
11 libclang-cpp.11.dylib    0x000000010b557aef (anonymous namespace)::DeferredDiagnosticsEmitter::checkFunc(clang::SourceLocation, clang::FunctionDecl*) + 1695
12 libclang-cpp.11.dylib    0x000000010b54dbbb clang::Sema::emitDeferredDiags() + 379
13 libclang-cpp.11.dylib    0x000000010b54ce40 clang::Sema::ActOnEndOfTranslationUnitFragment(clang::Sema::TUFragmentKind) + 576
14 libclang-cpp.11.dylib    0x000000010b54de59 clang::Sema::ActOnEndOfTranslationUnit() + 345
15 libclang-cpp.11.dylib    0x000000010af60de3 clang::Parser::ParseTopLevelDecl(clang::OpaquePtr<clang::DeclGroupRef>&, bool) + 1267
16 libclang-cpp.11.dylib    0x000000010aea470d clang::ParseAST(clang::Sema&, bool, bool) + 525
17 libclang-cpp.11.dylib    0x000000010c7865b8 clang::FrontendAction::Execute() + 104
18 libclang-cpp.11.dylib    0x000000010c717721 clang::CompilerInstance::ExecuteAction(clang::FrontendAction&) + 2145
19 libclang-cpp.11.dylib    0x000000010c7f806e clang::ExecuteCompilerInvocation(clang::CompilerInstance*) + 1998
20 clang-11                 0x0000000106d1fd53 cc1_main(llvm::ArrayRef<char const*>, char const*, void*) + 2131
21 clang-11                 0x0000000106d1de35 ExecuteCC1Tool(llvm::SmallVectorImpl<char const*>&) + 357
22 libclang-cpp.11.dylib    0x000000010c3b26c7 void llvm::function_ref<void ()>::callback_fn<clang::driver::CC1Command::Execute(llvm::ArrayRef<llvm::Optional<llvm::StringRef> >, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >*, bool*) const::$_1>(long) + 23
23 libLLVM-11.dylib         0x0000000112c21fda llvm::CrashRecoveryContext::RunSafely(llvm::function_ref<void ()>) + 218
24 libclang-cpp.11.dylib    0x000000010c3b1a65 clang::driver::CC1Command::Execute(llvm::ArrayRef<llvm::Optional<llvm::StringRef> >, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >*, bool*) const + 405
25 libclang-cpp.11.dylib    0x000000010c37ecf1 clang::driver::Compilation::ExecuteCommand(clang::driver::Command const&, clang::driver::Command const*&) const + 433
26 libclang-cpp.11.dylib    0x000000010c37f26c clang::driver::Compilation::ExecuteJobs(clang::driver::JobList const&, llvm::SmallVectorImpl<std::__1::pair<int, clang::driver::Command const*> >&) const + 140
27 libclang-cpp.11.dylib    0x000000010c398dec clang::driver::Driver::ExecuteCompilation(clang::driver::Compilation&, llvm::SmallVectorImpl<std::__1::pair<int, clang::driver::Command const*> >&) + 428
28 clang-11                 0x0000000106d1d7c8 main + 10264
29 dyld                     0x0000000107f8e4fe start + 462
clang-11: error: clang frontend command failed due to signal (use -v to see invocation)
clang version 11.0.0
Target: x86_64-apple-darwin13.4.0
Thread model: posix
InstalledDir: /Users/xinchen/miniconda3/envs/moose/bin
clang-11: note: diagnostic msg: Error generating preprocessed source(s).
make: *** [/Users/xinchen/projects/moose/modules/tensor_mechanics/build/unity_src/materials_crystal_plasticity_Unity.x86_64-apple-darwin13.4.0.opt.lo] Error 1

**Comment by GiudGiud** [comment, marked as answer]
Hello

Is this a regular or M1 mac?
What XCode did you have before? And you have 13.1 now?
Did you make clobberall before re-compiling?

Guillaume

**Reply by xchengood** [reply]
it is a regular (intel) mac.
Xcode that I had before is 12.4. later I update it to the newest one (13.3). when those errors appeared, I downgrade it to 13.1. But still has those errors.
Yes, I close all before re-trying.

**Reply by GiudGiud** [reply]
`make clobberall` cleans up the repository, it's not just closing the terminal.
13.3 is supported.

**Reply by xchengood** [reply]
Thank you. I did **make clobberall** before compiling and got below.

> 31 libclang-cpp.11.dylib    0x0000000112b6d26c clang::driver::Compilation::ExecuteJobs(clang::driver::JobList const&, llvm::SmallVectorImpl<std::__1::pair<int, clang::driver::Command const*> >&) const + 140
32 libclang-cpp.11.dylib    0x0000000112b86dec clang::driver::Driver::ExecuteCompilation(clang::driver::Compilation&, llvm::SmallVectorImpl<std::__1::pair<int, clang::driver::Command const*> >&) + 428
33 clang-11                 0x000000010e70e7c8 main + 10264
34 dyld                     0x00000001157e54fe start + 462
clang-11: error: clang frontend command failed due to signal (use -v to see invocation)
clang version 11.0.0
Target: x86_64-apple-darwin13.4.0
Thread model: posix
InstalledDir: /Users/xinchen/miniconda3/envs/moose/bin
20 errors generated.
make: *** [/Users/xinchen/projects/moose/framework/contrib/pcre/src/pcre_stringpiece.x86_64-apple-darwin13.4.0.opt.lo] Error 1
make: *** Waiting for unfinished jobs....
clang-11: note: diagnostic msg: 
********************

PLEASE ATTACH THE FOLLOWING FILES TO THE BUG REPORT:
Preprocessed source(s) and associated run script(s) are located at:
clang-11: note: diagnostic msg: /var/folders/4f/q1yn9fnd3plchfxkcg2bbj000000gn/T/pcrecpp-159296.cpp
clang-11: note: diagnostic msg: /var/folders/4f/q1yn9fnd3plchfxkcg2bbj000000gn/T/pcrecpp-159296.sh
clang-11: note: diagnostic msg: Crash backtrace is located in
clang-11: note: diagnostic msg: /Users/xinchen/Library/Logs/DiagnosticReports/clang-11_<YYYY-MM-DD-HHMMSS>_<hostname>.crash
clang-11: note: diagnostic msg: (choose the .crash file that corresponds to your crash)
clang-11: note: diagnostic msg: 

********************
make: *** [/Users/xinchen/projects/moose/framework/contrib/pcre/src/pcrecpp.x86_64-apple-darwin13.4.0.opt.lo] Error 1

**Reply by GiudGiud** [reply]
`make clobberall` failed?
If so your environment is borked. I d remove the conda environment and install again. Please us the new instructions using mamba  instead of conda too

**Reply by xchengood** [reply]
no, "make clobberall" works. after using that, I recompile and got that error.
ok, I will try to reinstall everything using mamba.
Thank you.

**Reply by xchengood** [reply]
Hi @GiudGiud. I uninstall Conda MOOSE Environment and install everything according to instruction using mamba. Then tried to compile but got the following warning and errors.
<img width="1414" alt="Screen Shot 2022-04-04 at 1 51 19 PM" src="https://user-images.githubusercontent.com/74226455/161612195-8256bcb2-8659-467d-8473-6cf64ca8c343.png">

**Reply by GiudGiud** [reply]
so the moose and libmesh are not in synch. 
Can you update the version of moose (from github)?
What does mamba list | grep moose return?

**Reply by xchengood** [reply]
I use "mamba update --all" to update. Is what I did correct?
after using "mamba list | grep moose", I got below. Then recompile, errors above are still there.
<img width="1089" alt="Screen Shot 2022-04-04 at 2 21 46 PM" src="https://user-images.githubusercontent.com/74226455/161617091-48493567-9e27-4264-8f24-3783d23e7a6e.png">

**Reply by GiudGiud** [reply]
it's correct but it only updates moose dependencies, not moose. 
To update moose it's
```
git fetch origin
git pull origin
```
if you are on the devel or master branch.

If you are on your own branch
```
git fetch origin
git rebase origin/devel
```

**Reply by xchengood** [reply]
Thank you. All issues are disappeared.
